near_deduplication/README.md

```diff
@@ -0,0 +1,126 @@
+# Near deduplication
+
+This is our implementation of near deduplication for BigCode dataset. It is largely evolved from the [original repo](https://github.com/bigcode-project/bigcode-dataset).
+
+### Setup
+
+````
+pip install -r requirements.txt
+````
+
+Login to be able to be able to push the dataset to the hub after deduplication and clone your huggingface-hub repositories:
+
+````
+huggingface-cli login
+````
+
+And make sure you have git-lfs installed.
+
+### Usage
+
+```bash
+# Quick example
+python minhash_deduplication_alt.py --dataset codeparrot/codeparrot-clean-valid \  
+    --split train \
+    --column content \
+    --cache-dir .cache \
+    --verbose
+# For details on the arguments, see the help message
+python minhash_deduplication_alt.py --help
+```
+
+Spark Script
+
+```bash
+# Upload the dataset to a bigquery table first!
+export CLUSTER_NAME=chenghao-temp
+export PROJECT_ID=huggingface-science-codeparrot
+export REGION=us-central1
+export ZONE=us-central1-a
+
+gcloud dataproc clusters create $CLUSTER_NAME \
+    --enable-component-gateway \
+    --region $REGION \
+    --zone $ZONE \
+    --master-machine-type c2d-standard-16 \
+    --master-boot-disk-size 500 \
+    --num-workers 10 \
+    --worker-machine-type c2d-standard-16 \
+    --worker-boot-disk-size 500 \
+    --image-version 2.0-debian10 \
+    --project $PROJECT_ID
+
+gcloud dataproc jobs submit pyspark --cluster ${CLUSTER_NAME} \
+    --region $REGION \
+    --jars gs://spark-lib/bigquery/spark-bigquery-latest_2.12.jar \
+    --driver-log-levels root=WARN \
+    --properties="spark.executor.memory"="50g","spark.driver.memory"="8g","spark.executor.cores"="14" \
+    near_deduplication/minhash_deduplication_spark.py \
+    -- \
+    --table "huggingface-science-codeparrot.the_stack_java.java" \
+    --output "gs://chenghao-data/dataproc_output/deduplicated"
+```
+
+With above settings, it took about 40 minutes to deduplicate the Java subset, 15x faster than the following python implementation in a comparable single-machine environment.
```

Comment:
That's amazing 🎉 !!
Maybe we can indicate number of files and volume (GB) of the Java subset, it doesn't necessarily generalize to other datasets but might be helpful to get an order of magnitude without needing to read the tables below.